esmvaltool/cmorizers/data/formatters/datasets/esacci_cloud.py

```diff
@@ -0,0 +1,404 @@
+                                ifile,
+                                inum,
+                                short_name,
+                                var,
+                                cmor_info,
+                                cubes,
+                                cubes_day,
+                            )
+                        except Exception as e:
+                            logger.error(
+                                "Error processing file '%s': %s", ifile, e
+                            )
+                else:
+                    _handle_missing_day(
+                        year, month, iday, short_name, cubes, cubes_day
+                    )
+
+        _concatenate_and_save_daily_cubes(
+            short_name, var, cubes, cubes_day, out_dir, cfg, cmor_info
+        )
+
+
+def _extract_variable_monthly(
+    short_name, var, cfg, in_dir, out_dir, start_date, end_date
+):
+    """Extract monthly variable with improved handling for multiple cubes."""
+    cmor_info = cfg["cmor_table"].get_variable(var["mip"], short_name)
+
+    if not start_date:
+        start_date = datetime(cfg["start_year_monthly"], 1, 1)
+    if not end_date:
+        end_date = datetime(cfg["end_year_monthly"], 12, 31)
+
+    cubes_am = iris.cube.CubeList()
+    cubes_pm = iris.cube.CubeList()
+
+    for year in range(start_date.year, end_date.year + 1):
+        for month in range(1, 13):  # Loop through all months (1-12)
+            # Construct the file list for the current month
+            filelist = glob.glob(
+                os.path.join(in_dir, f"{year}{month:02}" + var["file"])
+            )
+
+            if not filelist:
+                logger.warning(
+                    "No monthly file found for %s-%02d", year, month
+                )
+                continue
+
+            for ifile in filelist:
+                try:
+                    _process_monthly_file(
+                        ifile, short_name, var, cmor_info, cubes_am, cubes_pm
+                    )
+                except Exception as e:
+                    logger.error("Error processing file '%s': %s", ifile, e)
+
+    if cubes_am:
+        _concatenate_and_save_monthly_cubes(
+            short_name, var, cubes_am, out_dir, "-AM", cfg, cmor_info
+        )
+
+    if cubes_pm:
+        _concatenate_and_save_monthly_cubes(
+            short_name, var, cubes_pm, out_dir, "-PM", cfg, cmor_info
+        )
+
+    if cubes_am and cubes_pm:
+        # change day value in cubes_pm for concatinating
+        for cube in cubes_am:
+            time_coord = cube.coord("time")
+            new_time_points = [tpoint + 1 for tpoint in time_coord.points]
+            time_coord.points = new_time_points
+
+        cubes_combined = cubes_am + cubes_pm
+        _concatenate_and_save_monthly_cubes(
+            short_name, var, cubes_combined, out_dir, "-AMPM", cfg, cmor_info
+        )
+
+    if not (cubes_am or cubes_pm):
+        logger.warning("No valid cubes processed")
```

Comment:
Is this something you expect? If not, raise exception

Reply:
Done.